rest/rest.go

```diff
@@ -53,7 +54,18 @@ type EndpointType client.EndpointType
 type Server struct {
 	CoreAPI     bool
 	Protocol    string
-	Address     types.AddrPort
+	Address     *types.AddrPort
 	Certificate *shared.CertInfo
 	Resources   []Resources
 }
+
+// ValidateServerConfigs checks that the server configuration is valid.
+func (s Server) ValidateServerConfigs() error {
+	if s.CoreAPI {
+		if s.Address != nil || s.Protocol != "" || s.Certificate != nil {
+			return fmt.Errorf("Core API server cannot have Address, Protocol or Certificate")
+		}
```

Comment:
So it seems this PR will also allow an upstream project to specify their own path prefix for core endpoints, am I right about that?

If so, then I think we need some more validation there. I have a couple questions:
* Should we allow multiple sets of resources for the core API, or verify that the list is exactly 1 entry long?
* If we should allow multiple sets of resources, should they all have the same path prefix?
* Should we enforce that `Resources.Path` must be set? It's ambiguous what happens right now if it's unset.

Reply:
Yes, an upstream project can now define its own path prefix for core endpoints. To extend on this point a bit, since `extendedEndpoints` together with `resources.ExtendedEndpoints` is now removed, an upstream project would have to define its own path prefixes. The mental model here is to give more flexibility to the upstream project in shaping its own core routing structure.

Agreed on more validation is required, here are my thoughts let me know if you agree :slightly_smiling_face: 
- We should allow multiple sets of resources since we want to allow an upstream project to also define endpoints that does not have the `/1.0` prefix. Perhaps the restriction here should be only one `Server` with `CoreAPI=true` is allowed? I think we can be opinionated here since every `Server` with `CoreAPI=true` will be assigned to the same network listener anyway.
- I don't think we should enforce same path prefix at the `Resource` level if multiple resources are defined. A `Resource` should represent a logical grouping of endpoints that sit behind a path prefix. A relevant validation here could be to ensure duplicated path prefixes does not exist, this will also ensure no duplicated core endpoints exist.
- If `Resources.Path` is not set, then the path prefix becomes `/` which can be used to setup public endpoints like `/ui`, `/oidc` etc. I do agree that this is not immediately clear, so maybe we can enforce that `Resources.Path` must be set and if root prefix is required, it should be set to `Resources.Path="/"`? We would have to adjust [`internal.rest.HandleEndpoint`](https://github.com/canonical/microcluster/blob/df5dbb210195b8053b4940677350aa5cbe909897/internal/rest/rest.go#L152) to cater for this.

Reply:
- On point 2 I meant resources in a server. Like you mentioned we should allow same endpoints across different network listeners.
- Agreed on point 3, it's clearer to rename the field instead of a special case. I have validated the behaviour of setting `""` for the path prefix in this site manager [PR](https://github.com/canonical/lxd-site-manager/pull/15) (tested locally).